datalad/support/annexrepo.py

```diff
@@ -1726,12 +1730,48 @@ def _get_to_be_added_recs(paths):
         else:
             if backend:
                 options.extend(('--backend', backend))
-            for r in self._call_annex_records(
+            yield from self._maybe_batched_add(
+                files, options, jobs, expected_additions, batch=None)
+
+    def _maybe_batched_add(self, files, options, jobs, expected_additions, batch=None):
+        # Common code to be reused across existing implementations of "annex add" invocation
+        do_batch = False
+        if True: #  TEMP: always batch! batch or (batch is None and len(files) > 1):
+            # workaround to not use batch on directory paths, see:
+            # https://git-annex.branchable.com/todo/make_add_--batch_add_directories_content/ :
+            # We do use directories in tests directly but if used from higher level interfaces,
+            # we are unlikely to be passed folders here
+            do_batch = not any(isdir(opj(self.path, f)) for f in files)
+            if batch and not do_batch:  # batch mode requested but could not be used since we got directories!
+                lgr.debug("Not using git annex --batch since was given directories and "
+                          "git-annex add --batch does not support that yet ")
+
+        if do_batch:
+            # TODO: interfacing total_nbytes etc
+            # Note: took the same-ish 144s here and 169s on master for AnnexRepo only of
+            # python -m nose -s -v datalad/support/tests/test_annexrepo.py:test_files_split
+            #
+            # It seems that BatchedCommand lost `yield_` in
+            # 9f066efa6a0b316a133b6d6d0d4e4c2e1b856ce0 and ability to be used as a
```

Comment:
`yield_` was removed because it was not used anymore. It is no problem to bring it back, the `proc1`-method still exists and exhibits the original semantics.

Reply:
I have no immediate other requirements -- just want to have a generator to avoid the overhead of pre/post logic which we otherwise invoke per each file.  Do you think you could provide needed changes within this PR @christian-monch ?

Reply:
ping @christian-monch on this elderly request -- I am back to this PR

Reply:
Looking into the required changes